### PR TITLE
Js app service lib name unique

### PR DESCRIPTION
**Pull request by jsingle**
@thieman add validation to ensure that apps do not share the same name as services or libs

### CODE REVIEW

File: dusty/commands/validate.py [hunk 1]
```diff
@@ -37,7 +37,19 @@ def _validate_lib_references(lib, specs):
...
+def _check_name_overlap(specs):
+    apps = set(specs['apps'].keys())
+    libs = set(specs['libs'].keys())
+    services = set(specs['services'].keys())
+    app_service_overlap = apps.intersection(services)
+    if app_service_overlap:
+        raise ValidationException('Apps and services cannot share names: {}'.format(app_service_overlap))
+    app_lib_overlap = apps.intersection(libs)
+    if app_lib_overlap:
+        raise ValidationException('Apps and libs cannot share names: {}'.format(apps_lib_overlap))
```
Comment (thieman) [inline dusty/commands/validate.py]: Don't we just want to make all names unique across all types of specs?

Comment (jsingle) [inline dusty/commands/validate.py]: Well I'd say it makes sense to permit bundles to share names with apps.  We don't strictly need to check anything more than this to ensure functionality

Comment (thieman) [inline dusty/commands/validate.py]: Services and libs sharing names are ok? Seems weird.

**Comment by jsingle** [comment]
Services and libs seems weird but I didn't think it was worth explicitly checking it and raising an exception.  Don't mind fixing that though

**Comment by thieman** [comment]
Figure we might as well check it.

**Pull request merged**